cfme/cloud/availability_zone.py

```diff
@@ -1,72 +1,171 @@
+    save = Button('Save')
+    reset = Button('Reset')
+    cancel = Button('Cancel')
 
+    @property
+    def is_displayed(self):
+        return (
+            self.in_availability_zones and
+            self.title.text == 'Tag Assignment' and
+            '{} (Summary)'.format(self.context['object'].name) in self.breadcrumb.locations)
+        # TODO Add quadicon check to this return
 
-class CloudAvailabilityZoneTimelinesView(TimelinesView, BaseLoggedInPage):
+
+class CloudAvailabilityZoneTimelinesView(TimelinesView, AvailabilityZoneView):
     @property
     def is_displayed(self):
-        return self.logged_in_as_current_user and \
-            self.navigation.currently_selected == ['Compute', 'Clouds', 'Availability Zones'] and \
-            super(TimelinesView, self).is_displayed
+        return (
+            self.in_availability_zones and
+            self.breadcrumb.active_location == 'Timelines' and
+            "{} (Summary)".format(self.context['object'].name) in self.breadcrumb.locations and
+            super(TimelinesView, self).is_displayed)
 
 
 class AvailabilityZone(Navigatable):
-    def __init__(self, name, provider, appliance):
+    def __init__(self, name, provider, appliance=None):
         self.name = name
         self.provider = provider
         Navigatable.__init__(self, appliance=appliance)
 
 
 @navigator.register(AvailabilityZone, 'All')
 class AvailabilityZoneAll(CFMENavigateStep):
+    VIEW = AvailabilityZoneAllView
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
-    def am_i_here(self):
-        match_page(summary='Availability Zones')
-
     def step(self, *args, **kwargs):
         self.prerequisite_view.navigation.select('Compute', 'Clouds', 'Availability Zones')
 
 
 @navigator.register(AvailabilityZone, 'Details')
 class AvailabilityZoneDetails(CFMENavigateStep):
+    VIEW = AvailabilityZoneDetailsView
     prerequisite = NavigateToSibling('All')
 
-    def am_i_here(self):
-        match_page(summary='{} (Summary)'.format(self.obj.name))
-
     def step(self, *args, **kwargs):
-        tb.select('List View')
-        sel.click(listview_pagetable.find_row_by_cell_on_all_pages(
-            {'Name': self.obj.name,
-             'Cloud Provider': self.obj.provider.name}))
+        self.prerequisite_view.toolbar.view_selector.select('List View')
+        try:
+            row = self.prerequisite_view.paginator.find_row_on_pages(
+                self.prerequisite_view.entities.table,
+                name=self.obj.name,
+                cloud_provider=self.obj.provider.name)
+        except NoSuchElementException:
+            raise AvailabilityZoneNotFound('Could not locate Availability Zone "{}" on provider {}'
+                                           .format(self.obj.name, self.obj.provider.name))
+        row.click()
 
 
 @navigator.register(AvailabilityZone, 'EditTags')
 class AvailabilityZoneEditTags(CFMENavigateStep):
+    VIEW = AvailabilityZoneEditTagsView
```

Comment:
we will likely merge some widely used views when finish with main widgetastic conversion.